projects/cli/src/codegen/src/Commands/Generate.elm

```diff
@@ -1413,6 +1636,351 @@ toRoutePathToStringBranch file =
+                        , branches =
+                            [ { name = "Nothing"
+                              , arguments = []
+                              , expression =
+                                    CodeGen.Expression.value "Dict.empty"
+                              }
+                            , { name = "Just"
+                              , arguments = [ CodeGen.Argument.new "query" ]
+                              , expression =
+                                    CodeGen.Expression.ifElse
+                                        { condition =
+                                            CodeGen.Expression.function
+                                                { name = "String.isEmpty"
+                                                , arguments = [ CodeGen.Expression.value "query" ]
+                                                }
+                                        , ifBranch =
+                                            CodeGen.Expression.value "Dict.empty"
+                                        , elseBranch =
+                                            CodeGen.Expression.pipeline
+                                                [ CodeGen.Expression.value "query"
+                                                , CodeGen.Expression.function
+                                                    { name = "String.split"
+                                                    , arguments = [ CodeGen.Expression.string "&" ]
+                                                    }
+                                                , CodeGen.Expression.function
+                                                    { name = "List.filterMap"
+                                                    , arguments =
+                                                        [ CodeGen.Expression.parens
+                                                            [ CodeGen.Expression.function
+                                                                { name = "String.split"
+                                                                , arguments = [ CodeGen.Expression.string "=" ]
+                                                                }
+                                                            , CodeGen.Expression.operator ">>"
+                                                            , CodeGen.Expression.value "queryPiecesToTuple"
+                                                            ]
+                                                        ]
+                                                    }
+                                                , CodeGen.Expression.value "Dict.fromList"
+                                                ]
+                                        }
+                              }
+                            ]
+                        }
+                }
+            , CodeGen.Declaration.function
+                { name = "queryPiecesToTuple"
```

Comment:
This was taken from the existing `Route.Query` without much though, but I think most of it would be better replaced with [`Url.Builder.toQuery`](https://package.elm-lang.org/packages/elm/url/latest/Url-Builder#toQuery)